amgut/lib/data_access/redcap.py

```diff
@@ -0,0 +1,115 @@
+from urllib import urlencode
+from json import loads
+
+from tornado.web import HTTPError
+from tornado.httpclient import HTTPClient
+
+from amgut.lib.config_manager import AMGUT_CONFIG
+from amgut.lib.data_access.sql_connection import TRN
+
+
+def _make_request(data):
+    body = urlencode(data)
+    client = HTTPClient()
+    response = client.fetch(AMGUT_CONFIG.redcap_url, method='POST',
+                            headers=None, body=body)
+    # Response not always JSON, so can't always use loads
+    if '{"error"' in response.body:
+        raise HTTPError(400, loads(response.body)['error'])
+    return response.body
+
+
+def get_survey_url(record, instrument='ag_human_en_us'):
+    """Genereates a new survey link for a given record
+
+    Parameters
+    ----------
+    record : int
+        Redcap record (consent) to attach this survey to
+    instrument : str, optional
+        The instrument (survey) to get the URL for. Default ag_human_en_us
+
+    Returns
+    -------
+    str
+        URL for the survey
+    """
+    with TRN:
```

Comment:
That block is not in place because people can start the survey but not complete it, leaving an event id in limbo. The only scenario I can think of where someone would have a duplicate event_id is if someone has two browsers open and logged into the same person. This call is made internally in place of the original survey start, and automatically redirects to the survey URL. If blocking the event ID is a concern, I can try and come up with something that clears blocked event_ids after a couple hours or so,  but this may cause a leapfrog effect where someone starts a survey, then doesn't finish it, then starts a survey again before the clear and loses the blocked event_id.